### ISSUE TITLE
Build error against GCC 6.4.0

### ISSUE THREAD

**Issue by devrimgunduz** [pgbackrest/pgbackrest#545]
Hi,

I'm getting this error while building 2.03 on a machine that has GCC 6.4.0:

`gcc -I. -I../libc -std=c99 -O2 -Wfatal-errors -Wall -Wextra -Wwrite-strings -Wswitch-enum -Wconversion -Wformat=2 -Wformat-nonliteral -Wno-clobbered -Wno-missing-field-initializers `perl -MExtUtils::Embed -e ccopts` -DNDEBUG  -c command/archive/common.c -o command/archive/common.o
In file included from command/archive/common.c:8:0:
./command/archive/common.h:39:73: error: unknown type name ‘uint’
 ng *walSegmentNext(const String *walSegment, size_t walSegmentSize, uint pgVersion);
`                    
Any idea how I can fix?

Regards, Devrim

**Comment by dwsteele** [comment]
Yeah, `uint` turned out to be not as portable as I had hoped.  You can replace instances of `uint` with `unsigned int` if you just want to get it working.  I'll also remove these for the next release.

**Comment by devrimgunduz** [comment]
Hi @dwsteele ,

Thanks for the quick response.

As a non-hacker, I'd better wait for the patch, because a blind search/replace did not help :)

Thanks!

Regards, Devrim

**Comment by dwsteele** [comment]
I did the replacement and committed it at 3793ae1e4f0a4deeb156fa3d0 in master.  Can you give that a try and see how it goes?

**Comment by devrimgunduz** [comment]
Hi David,

Thanks, that error is gone. We have another one:

`gcc -I. -I../libc -std=c99 -O2 -Wfatal-errors -Wall -Wextra -Wwrite-strings -Wswitch-enum -Wconversion -Wformat=2 -Wformat-nonliteral -Wno-clobbered -Wno-missing-field-initializers `perl -MExtUtils::Embed -e ccopts` -DNDEBUG  -c perl/exec.c -o perl/exec.o
In file included from /usr/lib64/perl5/5.24.3/x86_64-linux/CORE/perl.h:28:0,
                 from ./perl/libc.auto.c:44,
                 from perl/exec.c:28:
/usr/lib64/perl5/5.24.3/x86_64-linux/CORE/config.h:1166:20: error: unknown type name ‘sigjmp_buf’
 #define Sigjmp_buf sigjmp_buf
                    ^
/usr/lib64/perl5/5.24.3/x86_64-linux/CORE/cop.h:34:5: note: in expansion of macro ‘Sigjmp_buf’
     Sigjmp_buf  je_buf;  /* uninit if je_prev is NULL */
     ^~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make: *** [Makefile:241: perl/exec.o] Error 1
`
Regards, Devrim

**Comment by dwsteele** [comment]
That one appears to be in a Perl header.  What OS/Version are you building on?

**Comment by devrimgunduz** [comment]
Hi David,

Well, don't ask :-) I had some spare time, and installed Gentoo after 15 years :-) Gentoo 2.4.1.

This is a part of the output from perl -V:

Characteristics of this binary (from libperl): 
  Compile-time options: HAS_TIMES PERLIO_LAYERS PERL_COPY_ON_WRITE
                        PERL_DONT_CREATE_GVSV
                        PERL_HASH_FUNC_ONE_AT_A_TIME_HARD PERL_MALLOC_WRAP
                        PERL_PRESERVE_IVUV USE_64_BIT_ALL USE_64_BIT_INT
                        USE_LARGE_FILES USE_LOCALE USE_LOCALE_COLLATE
                        USE_LOCALE_CTYPE USE_LOCALE_NUMERIC USE_LOCALE_TIME
                        USE_PERLIO USE_PERL_ATOF

**Comment by dwsteele** [comment]
What version of Perl is that?

**Comment by devrimgunduz** [comment]
5.24.3

**Comment by dwsteele** [comment]
I tried 5.24.1 on Debian 9 and didn't get any errors.  We have see this error before on RHEL 7 but never has a resolution since they started using packages: #501.

This actually seems to come up a lot and I found this helpful thread: https://www.paraview.org/pipermail/paraview-developers/2017-March/005199.html.

TLDR, try making with `CSTD=gnu99`. We use `c99` by default.

**Comment by dwsteele** [comment]
Please check the solution above (and the attached issue) and let us know if that works.

**Comment by devrimgunduz** [comment]
Hi David,

Just a quick update: It worked :)

Cheers,  Devrim
